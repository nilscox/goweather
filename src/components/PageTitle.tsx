/** @jsx jsx */
import { css, jsx } from '@emotion/core';

export type PageTitleProps = {
  /**
   * Title text
   */
  children: React.ReactNode,
};

/**
 * Display a bold centered text to be used as a title.
 */
const PageTitle = ({ children }: PageTitleProps) => (
  <h1 css={titleStyle} className="text-center py-4">
    { children }
  </h1>
);

const titleStyle = css`
  font-size: 20px;
  font-weight: bold;
`;

export default PageTitle;
