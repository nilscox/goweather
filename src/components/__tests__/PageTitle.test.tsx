import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import PageTitle from '../PageTitle';

test('PageTitle with text', () => {
  const tree = renderer
    .create(
      <Router>
        <PageTitle>Title with text</PageTitle>
      </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('PageTitle with node', () => {
  const tree = renderer
    .create(
      <Router>
        <PageTitle><span>span</span></PageTitle>
      </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
