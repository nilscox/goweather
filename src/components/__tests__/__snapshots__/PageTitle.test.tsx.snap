// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle with node 1`] = `
<h1
  className="text-center py-4 css-1aylpg"
>
  <span>
    span
  </span>
</h1>
`;

exports[`PageTitle with text 1`] = `
<h1
  className="text-center py-4 css-1aylpg"
>
  Title with text
</h1>
`;
